tests/storage/backendstorage.go

```diff
@@ -0,0 +1,131 @@
+ * limitations under the License.
+ *
+ * Copyright 2024 The KubeVirt Contributors
+ *
+ */
+
+package storage
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/pkg/pointer"
+	backendstorage "kubevirt.io/kubevirt/pkg/storage/backend-storage"
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/libvmifact"
+	"kubevirt.io/kubevirt/tests/util"
+)
+
+var _ = SIGDescribe("[Serial]Backend Storage", Serial, func() {
+	var virtClient kubecli.KubevirtClient
+
+	BeforeEach(func() {
+		virtClient = kubevirt.Client()
+	})
+
+	It("Should use the default storage class when none is specified in the CR", func() {
+		By("Removing the VM state storage class from the CR")
+		kv := util.GetCurrentKv(virtClient)
+		kv.Spec.Configuration.VMStateStorageClass = ""
+		tests.UpdateKubeVirtConfigValueAndWait(kv.Spec.Configuration)
+
+		By("Creating a VMI with persistent TPM")
+		vmi := libvmifact.NewCirros()
+		vmi.Spec.Domain.Devices.TPM = &v1.TPMDevice{Persistent: pointer.P(true)}
+		vmi = tests.RunVMIAndExpectLaunch(vmi, 60)
+
+		By("Expecting the creation of a backend storage PVC with the default storage class")
+		pvc, err := virtClient.CoreV1().PersistentVolumeClaims(vmi.Namespace).Get(context.Background(), backendstorage.PVCForVMI(vmi), metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(pvc.Spec.StorageClassName).NotTo(BeNil())
+		storageClass, err := virtClient.StorageV1().StorageClasses().Get(context.Background(), *pvc.Spec.StorageClassName, metav1.GetOptions{})
+		Expect(storageClass.Annotations).To(HaveKeyWithValue("storageclass.kubernetes.io/is-default-class", "true"))
+	})
+
+	// TODO: create a RequiresRWXFilesystemStorage decorator to remove the skip?
+	It("Should use RWO when RWX is not supported", func() {
+		var storageClass string
+
+		By("Finding a storage class that only supports filesystem in RWO")
+		sps, err := virtClient.CdiClient().CdiV1beta1().StorageProfiles().List(context.Background(), metav1.ListOptions{})
+		Expect(err).NotTo(HaveOccurred()) // Assumes CDI is present. Should we also add a decorator for that?
+		for _, sp := range sps.Items {
+			fsrwo := false
+			fsrwx := false
+			for _, property := range sp.Status.ClaimPropertySets {
+				if property.VolumeMode != nil && *property.VolumeMode == corev1.PersistentVolumeFilesystem {
+					if property.AccessModes != nil {
+						for _, accessMode := range property.AccessModes {
+							if accessMode == corev1.ReadWriteMany {
+								fsrwx = true
+							}
+							if accessMode == corev1.ReadWriteOnce {
+								fsrwo = true
+							}
+						}
+					}
+				}
+			}
+			if fsrwo && !fsrwx && sp.Status.StorageClass != nil {
+				storageClass = *sp.Status.StorageClass
```

Comment:
Why can't we use the [`GetRWOFileSystemStorageClass`](https://github.com/kubevirt/kubevirt/blob/main/tests/libstorage/storageclass.go#L232) function?

Reply:
I don't think it guarantees the storage class doesn't support RWX... But if it does yeah I can definitely switch to it.